File: webstore/webstore-java/src/main/java/com/renan/webstore/dto/CategoryDTO.java
package com.renan.webstore.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class CategoryDTO {
    
    private Integer id;
    private String name;
    private List<CategoryDTO> subCategories;
    
}

File: webstore/webstore-kotlin/src/main/kotlin/com/renan/webstore/dto/CategoryDTO.kt
package com.renan.webstore.dto

import com.fasterxml.jackson.annotation.JsonInclude
import com.fasterxml.jackson.annotation.JsonInclude.Include

@JsonInclude(Include.NON_NULL)
data class CategoryDTO(
    val id: Int,
    val name: String,
    var subCategories: List<CategoryDTO>? = null
)
